assets/js/script.js

```diff
@@ -0,0 +1,146 @@
+    hasNumbers: generateNumber,
+    hasSymbols: generateSymbols
+}
+
+/*
+
+Click Event
+
+*/
+
+// Add event listener to check password options, and generate a password based on true options
+
+generateBtn.addEventListener('click', () => {
+
+    // check length for password, set by user
+    const length = passwordLength.value;
+
+    // check checked values of password options, set by user
+    const hasLowercase = lowercaseCheck.checked;
+    const hasUpercase = uppercaseCheck.checked;
+    const hasNumbers = numbersCheck.checked;
+    const hasSymbols = symbolsCheck.checked;
+
+    // parameters to be passed
+    // pass checked values into generatePassword function
+    // pass generatedPassword() into the password DOM Element
+    passwordText.value = generatePassword(length, hasLowercase, hasUpercase, hasNumbers, hasSymbols);
+
+})
+
+/*
+
+Functions
+
+*/
+
+function generatePassword(length, hasLowercase, hasUpercase, hasNumbers, hasSymbols) {
+
+    // string to store generated password
+    let pwd = ''
+
+    // parameters being passed from addEventListener
+    // count how many options are checked
+    const countCheck = hasLowercase + hasUpercase + hasNumbers + hasSymbols;
+
+    // array of objects
+    // get values of each checked box
+    // filter method to filters out values === false
+    const countArray = [{ hasLowercase }, { hasUpercase }, { hasNumbers }, { hasSymbols }].filter(check => Object.values(check)[0]);
+
+    // condition for length of password
+    if (length < 8) {
+        return alert("Please enter a password length which is greater than or equal to 8!");
+    }
+
+    // condition for length of password
+    if (length >= 128) {
+        return alert("Please enter a password length which is less than 128!");
+    }
+
+    // conditional statement for no option checked on page
+    if (countCheck === 0) {
+        return alert("Make sure to include at least one character option, to generate a password!");
+    }
+
+    // loop through password length
+    for (i = 0; i < length; i += countCheck) {
+
+        // on every loop
+        // loop on through each array type (character type)
+        countArray.forEach(type => {
+
+            // variable to store each key to run funtion
+            const characterType = Object.keys(type)[0];
+            // e.g. hasLowercase
+
+            // pass key into variable list of functions, to generate character
+            // store each character into pwd string
+            pwd += functionList[characterType]();
+        });
+    }
+
+    // shuffle generated password
+
+    // split string into array
+    const randomPassword = pwd.split("");
+
+    // pass array parameter into function
+    shufflePassword(randomPassword);
+
+    // rejoin shuffled characters into string
+    const finalPassword = randomPassword.join("");
+
+    return finalPassword;
+
+}
+
+// function to shuffle the generate password
+function shufflePassword(a) {
```

Comment:
It is best practice to use ES6 (arrow) functions. If you changed these functions to ES6 functions, it would have to be moved to above where the function is called as they are not hoisted like function declarations are. 
```suggestion
const shufflePassword = (a) => {
```